src/instamatic/processing/PETS_input_factory.py

```diff
@@ -0,0 +1,195 @@
+from __future__ import annotations
+
+from copy import deepcopy
+from dataclasses import dataclass
+from importlib import resources
+from textwrap import dedent
+from time import ctime
+from typing import Any, Iterable, Optional, Union
+from warnings import warn
+
+import pandas as pd
+from typing_extensions import Self
+
+from instamatic import config
+from instamatic._collections import partial_formatter
+from instamatic._typing import AnyPath
+
+
+class PetsKeywords:
+    """Read PETS2 field metadata as read from the PETS2 manual."""
+
+    def __init__(self, table: pd.DataFrame) -> None:
+        table.fillna({'end': 'optional'}, inplace=True)
+        self.table = table
+
+    @classmethod
+    def from_file(cls, path: AnyPath) -> Self:
+        return cls(pd.read_csv(path, index_col='field'))
+
+    @property
+    def list(self) -> list[str]:
+        return self.table.index.values.tolist()
+
+    def find(self, text: str) -> set[str]:
+        firsts = {t.strip().split()[0].lower() for t in text.splitlines() if t.strip()}
+        return firsts.intersection(self.list)
+
+
+with resources.files('instamatic.processing').joinpath('PETS_input_keywords.csv').open() as f:
+    pets_keywords = PetsKeywords.from_file(f)  # noqa: default keywords read at import
```

Comment:
Can you move this to the top of the file? Also capitalize pets keywords being a global.

```suggestion
with resources.files('instamatic.processing').joinpath('PETS_input_keywords.csv').open() as f:
    PETS_KEYWORDS = PetsKeywords.from_file(f)  # noqa: default keywords read at import
```

Reply:
As with `AnyPetsInputElement`, can't move it higher because it's an instance of a `PetsKeywords` defined immediately before; consequently it is as high as possible. I do agree about making it uppercase.